tools/ramdump/ramdump_tool.c

```diff
@@ -80,69 +91,201 @@ static int b_read(int fd, uint8_t *buf, int size)
+		printf("Please enter correct input: ");
+		scanf("%d", &index);
+	}
+
+	/* Mark regions to be dumped */
+	while (index > 0) {
+		if((index % 10) > (number_of_regions + 1)) {
+			index = -1;
+			goto scan_input;
 		}
+		mem_info[index % 10].rd_regionx_mark = 1;
+		index = index / 10;
+		regions_to_dump++;
+	}
 
-		count++;
-		ramdump_size--;
+	/* If any digit in input integer is 1, dump all regions */
+	if (mem_info[1].rd_regionx_mark) {
+		mem_info[1].rd_regionx_mark = 0;
+		regions_to_dump = number_of_regions;
+		for (i = 2; i <= number_of_regions + 1; i++) {
+			mem_info[i].rd_regionx_mark = 1;
+		}
+	}
 
-		if ((count % (16 * 1024)) == 0) {
-			// printf("%s: received %d bytes\n", __func__, count);
-			printf("\b=>");
+	/* Send number of regions to dump to TARGET */
+	snprintf(ramdump_region, 2, "%d", regions_to_dump);
+	ret = send_region_info(dev_fd, ramdump_region);
+	if (ret != 0) {
+		printf("Receiving number of regions to be dumped failed, ret = %d\n", ret);
+		return -1;
+	}
+	printf("\nNo. of Regions to be dumped received: %s\n", __func__);
+	printf("\nReceiving ramdump......\n\n");
+
+	/* Dump data region wise */
+	for (index = 2; index <= number_of_regions + 1; index++) {
+		if (mem_info[index].rd_regionx_mark == 1) {
+
+			/* Send region index to TARGET */
+			snprintf(ramdump_region, 2, "%d", index - 2);
+			ret = send_region_info(dev_fd, ramdump_region);
+			if (ret != 0) {
+				printf("Receiving region index failed, ret = %d\n", ret);
+				return -1;
+			}
+
+			snprintf(bin_file, BINFILE_NAME_SIZE, "ramdump_0x%08x_0x%08x.bin",  mem_info[index].rd_regionx_start, (mem_info[index].rd_regionx_start +  mem_info[index].rd_regionx_size));
+			printf("=========================================================================\n");
+			if (number_of_regions == 1) {
+				printf("Dumping data, Address: 0x%08x, Size: %dbytes\n",  mem_info[index].rd_regionx_start,  mem_info[index].rd_regionx_size);
+			} else {
+				printf("Dumping Region: %d, Address: 0x%08x, Size: %dbytes\n", index - 2,  mem_info[index].rd_regionx_start,  mem_info[index].rd_regionx_size);
+			}
+			printf("=========================================================================\n");
+
+			bin_fp = fopen(bin_file, "w");
+			if (bin_fp == NULL) {
+				printf("%s create failed\n", bin_file);
+				return -1;
+			}
+
+			printf("[>");
 			fflush(stdout);
+
+			/* Dump data of Memory REGIONx  */
+			ramdump_size =  mem_info[index].rd_regionx_size;
+			while (ramdump_size) {
+				ret = read(dev_fd, &buf, 1);
+				if (ret != 1) {
+					printf("Receiving ramdump %dTH byte failed, ret = %d\n", count, ret);
+					fclose(bin_fp);
+					return -1;
+				}
+
+				ret = fwrite(&buf, 1, 1, bin_fp);
+				if (ret != 1) {
+					printf("Writing ramdump %dTH byte failed, ret = %d\n", count, ret);
+					fclose(bin_fp);
+					return -1;
+				}
+
+				count++;
+				ramdump_size--;
+
+				if ((count % (16 * 1024)) == 0) {
```

Comment:
How about changing 16 and 1024 to meaningful defined value?

Reply:
done